### PR TITLE
Handle new transfers fields

### PR DESCRIPTION
**Pull request by pailakka**
Handle from_trip_id and to_trip_id fields in transfers.txt

### CODE REVIEW

**Comment by laurentg** [comment]
Travis CI fails:

> ../home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/sqlalchemy/sql/crud.py:793: SAWarning: Column 'transfers.from_trip_id' is marked as a member of the primary key for table 'transfers', but has no Python-side or server-side default generator indicated, nor does it indicate 'autoincrement=True' or 'nullable=True', and no explicit value is passed.  Primary key columns typically may not store NULL. Note that as of SQLAlchemy 1.1, 'autoincrement=True' must be indicated explicitly for composite (e.g. multicolumn) primary keys if AUTO_INCREMENT/SERIAL/IDENTITY behavior is expected for one of the columns in the primary key. CREATE TABLE statements are impacted by this change as well on most backends.

File: gtfslib/converter.py [hunk 1]
```diff
@@ -470,6 +451,25 @@ def import_stop(stop, stoptype, zone_ids, item_ids, station_ids=None):
...
     logger.info("Imported %d trips" % n_trips)
 
+    logger.info("Importing transfers...")
+    n_transfers = 0
+    for transfer in gtfs.transfers():
+        from_stop_id = transfer.get('from_stop_id')
+        to_stop_id = transfer.get('to_stop_id')
+        transfer['transfer_type'] = _toint(transfer.get('transfer_type'), 0)
+        for stop_id in (from_stop_id, to_stop_id):
+            if stop_id not in station_ids and stop_id not in stop_ids:
```
Comment (laurentg) [inline gtfslib/converter.py:461]: Maybe we should also do the same test for from/to trip IDs, to test if they exists, and skip the failing transfer in lenient mode or raise an error in strict mode.

**Comment by pailakka** [comment]
Thanks for your feedback! I realized that trip_ids in transfers.txt make transfers rather hard to model in a DB as to/from_trip_id is optional field and there could be many combinations of trip_ids for every pair of stop ids. To handle this, I added transfer_id autoincrement field to act as primary key. I'm not however sure if this is good enough solution?

File: gtfslib/converter.py (hunk 1 above)
Comment (pailakka) [reply gtfslib/converter.py:461]: I added these checks

**Comment by laurentg** [comment]
Great, thanks! Indeed I'm not a specialist too, having to define a multi-column (4 fields!) primary key with both non-null and null values seems a bit complex. I do not see any drawback on having an auto-increment ID column.

I assume you have tested your changes? I can merge it w/o testing myself. Would it be possible to include a few unit tests with transfer trip IDs?

**Comment by pailakka** [comment]
I'll do some more test and implement a few unit tests

**Pull request closed without merging**